apis/v1beta1/config.go

```diff
@@ -474,23 +474,81 @@ func (s *Service) ApplyDefaults(logger logr.Logger) error {
 		return err
 	}
 
-	tm := &AnyConfig{
-		Object: map[string]interface{}{
-			"metrics": map[string]interface{}{
-				"address": fmt.Sprintf("%s:%d", telemetryAddr, telemetryPort),
-			},
-		},
+	if s.Telemetry == nil {
+		s.Telemetry = &AnyConfig{
+			Object: map[string]any{},
+		}
 	}
 
-	if s.Telemetry == nil {
-		s.Telemetry = tm
+	metrics, ok := s.Telemetry.Object["metrics"]
+	if !ok {
+		metrics = map[string]any{}
+		s.Telemetry.Object["metrics"] = metrics
+	}
+	metricsMap, ok := metrics.(map[string]any)
+	if !ok {
+		logger.Info("metrics field is not a map")
+		return nil
+	}
+
+	readers, ok := metricsMap["readers"]
+	if !ok {
+		readers = []any{}
+		metricsMap["readers"] = readers
+	}
+	readersSlice, ok := readers.([]any)
+	if !ok {
+		logger.Info("readers field is not a slice")
 		return nil
 	}
-	// NOTE: Merge without overwrite. If a telemetry endpoint is specified, the defaulting
-	// respects the configuration and returns an equal value.
-	if err := mergo.Merge(s.Telemetry, tm); err != nil {
```

Comment:
I couldn't just use mergo.Merge here because it doesn't work when an element exists in the readers. It simply doesn't add the default endpoint in that case.